internal/brokerstore/brokerstore.go

```diff
@@ -294,3 +299,13 @@ func findAllMatchingKeyPrefix(mp *sync.Map, prefix string) []interface{} {
 	})
 	return values
 }
+
+// FlushAllProducers will iterate over the producer map, and flush all messages in the producer buffer
+func (b *BrokerStore) FlushAllProducers() bool {
+	b.producerMap.Range(func(key, producer interface{}) bool {
+		producer.(messagebroker.Producer).Flush(defaultFlushTimeoutMs)
```

Comment:
This is a sequential call. Which would add up if the num of producers are high. Please make it parallel.